router.js

```diff
@@ -57,30 +57,39 @@ Router.prototype.match = function (req) {
   return false;
 }
 
+var flatten = function (arr, ret) {
+  var ret = ret || [];
+  for (var i = 0, len = arr.length; i < len; ++i) {
+    if (Array.isArray(arr[i])) {
+      flatten(arr[i], ret);
+    } else {
+      ret.push(arr[i]);
+    }
+  }
+  return ret;
+}
+
 /**
  * Registers new route
  * @param method
  * @param path
- * @param callbacks
- * @param options
+ * @param callback
  */
-Router.prototype.add = function (method, path, callbacks, options) {
-  function flatten(arr, ret) {
-    var ret = ret || []
-      , len = arr.length;
-    for (var i = 0; i < len; ++i) {
-      if (Array.isArray(arr[i])) {
-        flatten(arr[i], ret);
-      } else {
-        ret.push(arr[i]);
-      }
-    }
-    return ret;
-  }
+Router.prototype.add = function (method, path, callback) {
+  var hasOptions = (typeof callback !== 'function' && !Array.isArray(callback));
+  var callbacks = [].slice.call(arguments, 2);
```

Comment:
Yes, I know and my first version was using [Bluebird workaround](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments), but even then function was marked as not optimized. Instead, I've used what was in the code:
https://github.com/visualfanatic/named-routes/blob/d24b2538213279cca0437224e8afbc30d36c1338/router.js#L312